File: 2025/jimmycho/week9/2025-08-03-Gas-Station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        offsets = [gas_val - cost_val for gas_val, cost_val in zip(gas, cost)]
        total, tank, start = 0, 0, 0
        for i in range(len(gas)):
            total += offsets[i]
            tank += offsets[i]
            if tank < 0:
                tank = 0
                start = i + 1
        return start if total >= 0 else -1

File: 2025/jiyeon/2025-08-03-Gas-Station.java
class GasStation {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int curGas = 0, totalGas = 0;
        int start = 0;
        for (int i = 0; i < gas.length; i++) {
            int now = gas[i] - cost[i];
            curGas += now;
            totalGas += now;
            if (curGas < 0) {
                curGas = 0;
                start = i + 1;
            }
        }

        return totalGas >= 0 ? start : -1;
    }
}